tests/sentry/relay/config/test_metric_extraction.py

```diff
@@ -1167,6 +1167,41 @@ def test_get_metrics_extraction_config_features_combinations(
             assert len(config["metrics"]) == number_of_metrics
 
 
+def _metric_spec(tags: list[dict[str, str]]) -> dict[str, Any]:
+    return {
+        "category": "transaction",
+        "condition": {"name": "event.duration", "op": "gte", "value": 10.0},
+        "field": None,
+        "mri": "c:transactions/on_demand@none",
+        "tags": tags,
+    }
+
+
+@django_db_all
+def test_include_environment_for_widgets(default_project: Project) -> None:
+    env_tag = {"field": "event.environment", "key": "environment"}
+    query = "transaction.duration:>=10"
+
+    with Feature([ON_DEMAND_METRICS, ON_DEMAND_METRICS_WIDGETS]):
+        create_widget(["count()"], query, default_project)
+        config = get_metric_extraction_config(default_project)
+        # Because we have two specs we will have two metrics.
+        # The second spec includes the environment tag as part of the query hash.
+        assert config and config["metrics"] == [
+            # _metric_spec([{"key": "query_hash", "value": "f1353b0f"}]),
+            _metric_spec([{"key": "query_hash", "value": "f1353b0f"}, env_tag]),
```

Comment:
Notice this env tag is included as part of the tags but it won't be the same down below.

Reply:
Yeah that's right, since widgets are `DYNAMIC_QUERY` and alerts are `SIMPLE_QUERY`